src/tools/profile/read_profile.c

```diff
@@ -131,49 +167,76 @@ static double time_to_usec(profile_data_t *data, options_t *opts, uint64_t time)
 
 static int compare_locations(const void *l1, const void *l2)
 {
-    const ucs_profile_location_t *loc1 = l1;
-    const ucs_profile_location_t *loc2 = l2;
+    const ucs_profile_thread_location_t *loc1 = l1;
+    const ucs_profile_thread_location_t *loc2 = l2;
     return (loc1->total_time > loc2->total_time) ? -1 :
            (loc1->total_time < loc2->total_time) ? +1 :
            0;
 }
 
-static void show_profile_data_accum(profile_data_t *data, options_t *opts)
+static int show_profile_data_accum(profile_data_t *data, options_t *opts)
 {
-    uint32_t num_locations = data->header->num_locations;
-    ucs_profile_location_t *sorted_locations;
-    ucs_profile_location_t *loc;
+    const uint32_t            num_locations          = data->header->num_locations;
+    profile_sorted_location_t *sorted_locations      = NULL;
+    const ucs_profile_thread_location_t *thread_location;
+    profile_sorted_location_t *sorted_loc;
+    const ucs_profile_location_t *loc;
+    unsigned location_idx;
+    int ret;
 
-    sorted_locations = malloc(sizeof(*sorted_locations) * num_locations);
+    sorted_locations = calloc(sizeof(*sorted_locations), num_locations);
```

Comment:
there is no difference, but to align with calloc arguments )1st arg - num entries, 2nd arg - size)
`calloc(num_locations, sizeof(*sorted_locations))`